test/module/irohad/ametsuchi/postgres_executor_test.cpp

```diff
@@ -1996,5 +1996,204 @@ namespace iroha {
       CHECK_ERROR_CODE_AND_MESSAGE(cmd_result, 7, query_args);
     }
 
+    class CompareAndSetAccountDetail : public CommandExecutorTest {
+     public:
+      void SetUp() override {
+        CommandExecutorTest::SetUp();
+        createDefaultRole();
+        createDefaultDomain();
+        createDefaultAccount();
+        account2_id = "id2@" + domain_id;
+        CHECK_SUCCESSFUL_RESULT(
+            execute(*mock_command_factory->constructCreateAccount(
+                        "id2",
+                        domain_id,
+                        shared_model::interface::types::PubkeyType(
+                            std::string('2', 32))),
+                    true));
+      }
+      shared_model::interface::types::AccountIdType account2_id;
+    };
+
+    /**
+     * @given command
+     * @when trying to set kv
+     * @then kv is set
+     */
+    TEST_F(CompareAndSetAccountDetail, Valid) {
+      CHECK_SUCCESSFUL_RESULT(
+          execute(*mock_command_factory->constructCompareAndSetAccountDetail(
+              account_id, "key", "value", boost::none)));
```

Comment:
I cannot find where are the relevant permissions added. if they are not added, why it succeeds then?

Reply:
Because account_id change his own details